include/nbl/system/IFile.h

```diff
@@ -8,9 +8,12 @@
 namespace nbl {
 namespace system
 {
-
 class IFile : public core::IReferenceCounted
 {
+	// Don't like this approach, gonna stack up many forward declarations, @criss any suggestions?
+	// I also had an idea to make interfaces public in the implementations of IFile
+	// and static_cast files before calling, but that would invalidate any other IFile descendants calls
+	friend class CSystemCallerWin32;
```

Comment:
i dont get it, whats the problem?
`IFile` must not know about any platform-specific stuff, get rid of this `friend`

Reply:
just `static_cast` in implementation classes, why would you need this `friend`?